### PR TITLE
chore(infra): streamline npm tasks

### PR DESCRIPTION
**Pull request by hgw77**
# Summary

this will PR will handle the know problem for concurrency and persistent tasks in turbo like:

````
invalid task configuration
# Error:   × You have 21 persistent tasks but `turbo` is configured for concurrency of 10. Set --concurrency to at least 22 
````

that issue only happens for the global dev task because they are persistent

# Changes Made

* created a new script `check-concurrency.sh`
* this script is used to check the persistent tasks and will set the concurrency dynamically from the default of 10 to the calculated value
* small adjustments for npm tasks

1. `npm i`
2. `npm run dev`

Note: at the moment this will not run to the end because some apps are not proper configured related to the ports that are used

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -8,15 +8,14 @@
       "cache": true
     },
     "lint": {
-      "dependsOn": ["^lint"],
```
Comment (andypf) [inline turbo.json:11]: 👍

File: turbo.json [hunk 2]
```diff
@@ -8,15 +8,14 @@
...
       "cache": true
     },
     "typecheck": {
       "cache": true
     },
     "test": {
       "cache": true,
-      "dependsOn": ["build"]
+      "dependsOn": ["^build"]
```
Comment (andypf) [inline turbo.json:18]: 👍

File: check-concurrency.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (hgw77) [reply check-concurrency.sh:1]: the dot (to source the variables) should also work in zsh,
and yes if we want to set dynamically the concurrency above the default of 10 we need that

**Review by ArtieReus — DISMISSED** [review]
works also in darwin:

```
┏D063222@WXL779PFV6 /Users/d063222/Documents/sap/cloudoperators/juno u8s:[qa-de-1|default] kubectl:[qa-de-1] git:(hgw-streamline-npm-tasks)
┗➜  npm run dev

> dev
> . ./check-concurrency.sh && turbo run dev --concurrency=$CONCURRENCY

turbo 2.1.2

persistent tasks: 24
set concurrency to: 25
╭──────────────────────────────────────────────────────────────────────────╮
│                                                                          │
│                     Update available v2.1.2 ≫ v2.1.3                     │
│    Changelog: https://github.com/vercel/turborepo/releases/tag/v2.1.3    │
│             Run "npx @turbo/codemod@latest update" to update             │
│                                                                          │
│          Follow @turborepo for updates: https://x.com/turborepo          │
╰──────────────────────────────────────────────────────────────────────────╯
```

**Review by hodanoori — APPROVED** [review]
I like what you have done to calculate concurrency number dynamically. Good job!

**Review by andypf — APPROVED** [review]
LGTM, good job

**Pull request merged**